lib/fortnox/api/repositories/base/json_convertion.rb

```diff
@@ -39,7 +39,16 @@ def default_key_from_json_transform( key )
 
         def convert_hash_keys_to_json_format( hash, key_map )
           hash.each_with_object( {} ) do |(key, value), json_hash|
-            json_hash[ convert_key_to_json( key, key_map ) ] = value
+            if value.is_a?(Array) # Is attribute a nested model?
+              nested_models = []
+              value.each do |nested_model|
+                nested_key_map = key_map.fetch( key, {} )
```

Comment:
What if you get a string back here, as in the parent key needs translating but not the nested model attributes?

Wouldn't it be better to base the recursion on how the  translations map looks? If it has a hash as the value for this key you know you need to recurse.

Right now you will error out if: 1) The parent key needs a translation but not the nested model. 2) The array is of atomic values and not objects, such as several ENUM values etc. 3) The nested model is singular, ie. not an array.